src/model/add-dots.ts

```diff
@@ -0,0 +1,56 @@
+// Wraps the last dot in a footer
+const transformLastDot = (element: string): string => {
+	const modified = `<footer>${element}</footer>`;
+	return modified.replace(
+		new RegExp('[•]', 'g'),
+		'<span data-dcr-style="bullet"></span>',
+	);
+};
+
+const checkForDots = (elements: CAPIElement[]): CAPIElement[] => {
+	// Loop over elements and check if a dot is in the TextBlockElement
+	const enhanced: CAPIElement[] = [];
+	elements.map((element, i) => {
+		if (
+			element._type ===
+				'model.dotcomrendering.pageElements.TextBlockElement' &&
+			element.html.includes('•')
+		) {
+			if (elements.length - 1 === i) {
+				enhanced.push({
+					...element,
+					html: transformLastDot(element.html),
+				});
+			} else {
+				enhanced.push({
+					...element,
+					html: element.html.replace(
+						new RegExp('[•]', 'g'),
+						'<span data-dcr-style="bullet"></span>',
+					),
+				});
+			}
+		} else {
+			enhanced.push(element);
+		}
+	});
+	return enhanced;
+};
+
+export const addDots = (data: CAPIType): CAPIType => {
+	data.standfirst = data.standfirst.replace(
+		new RegExp('[•]', 'g'),
+		'<span data-dcr-style="bullet"></span>',
+	);
```

Comment:
This is mutating (sorry! 😉 ). I've added a suggestion in the return statement to show how we can spread the change in.

PS. You know you're mutating when you take the original object (`data` in this case) and assign something to it, as in `data.thing = newVal`.